# YouTube Trending Video EDA CSV

In [1]:
import pandas as pd
import numpy as np
from matplotlib import pyplot as plt
import seaborn as sns

In [3]:
df = pd.read_csv("../../data/trends/USvideos.csv")
df.head(3)

video_id trending_date  \
0  2kyS6SvSYSE      17.14.11   
1  1ZAPwfrtAFY      17.14.11   
2  5qpjK5DgCt4      17.14.11   

                                               title    channel_title  \
0                 WE WANT TO TALK ABOUT OUR MARRIAGE     CaseyNeistat   
1  The Trump Presidency: Last Week Tonight with J...  LastWeekTonight   
2  Racist Superman | Rudy Mancuso, King Bach & Le...     Rudy Mancuso   

   category_id              publish_time  \
0           22  2017-11-13T17:13:01.000Z   
1           24  2017-11-13T07:30:00.000Z   
2           23  2017-11-12T19:05:24.000Z   

                                                tags    views   likes  \
0                                    SHANtell martin   748374   57527   
1  last week tonight trump presidency|"last week ...  2418783   97185   
2  racist superman|"rudy"|"mancuso"|"king"|"bach"...  3191434  146033   

   dislikes  comment_count                                  thumbnail_link  \
0      2966          15954  https://i.ytimg.com/vi/2kyS6SvSYSE/default.jpg   
1      6146          12703  https://i.ytimg.com/vi/1ZAPwfrtAFY/default.jpg   
2      5339           8181  https://i.ytimg.com/vi/5qpjK5DgCt4/default.jpg   

   comments_disabled  ratings_disabled  video_error_or_removed  \
0              False             False                   False   
1              False             False                   False   
2              False             False                   False   

                                         description  
0  SHANTELL'S CHANNEL - https://www.youtube.com/s...  
1  One year after the presidential election, John...  
2  WATCH MY PREVIOUS VIDEO ▶ \n\nSUBSCRIBE ► http...

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40949 entries, 0 to 40948
Data columns (total 16 columns):
video_id                  40949 non-null object
trending_date             40949 non-null object
title                     40949 non-null object
channel_title             40949 non-null object
category_id               40949 non-null int64
publish_time              40949 non-null object
tags                      40949 non-null object
views                     40949 non-null int64
likes                     40949 non-null int64
dislikes                  40949 non-null int64
comment_count             40949 non-null int64
thumbnail_link            40949 non-null object
comments_disabled         40949 non-null bool
ratings_disabled          40949 non-null bool
video_error_or_removed    40949 non-null bool
description               40379 non-null object
dtypes: bool(3), int64(5), object(8)
memory usage: 4.2+ MB


In [9]:
df[df["description"].apply(lambda x: pd.isna(x))].head(3)

video_id trending_date  \
42   NZFhMSgbKKM      17.14.11   
47   sbcbvuitiTc      17.14.11   
175  4d07RXYLsJE      17.14.11   

                                                 title  \
42   Dennis Smith Jr. and LeBron James go back and ...   
47   Stephon Marbury and Jimmer Fredette fight in C...   
175                Sphaera - demonstrating interaction   

                channel_title  category_id              publish_time  \
42               Ben Rohrbach           17  2017-11-13T15:11:00.000Z   
47   NBA Highlights · YouTube           17  2017-11-10T18:23:05.000Z   
175              Jenny Hanell           28  2017-11-04T20:48:16.000Z   

                          tags   views  likes  dislikes  comment_count  \
42                      [none]     945      7         5              8   
47   NBA|"Basketball"|"Sports"  956169   2017      2425           1447   
175                     [none]    1827      3         0              2   

                                     thumbnail_link  comments_disabled  \
42   https://i.ytimg.com/vi/NZFhMSgbKKM/default.jpg              False   
47   https://i.ytimg.com/vi/sbcbvuitiTc/default.jpg              False   
175  https://i.ytimg.com/vi/4d07RXYLsJE/default.jpg              False   

     ratings_disabled  video_error_or_removed description  
42              False                   False         NaN  
47              False                   False         NaN  
175             False                   False         NaN

In [10]:
df["description"] = df["description"].fillna(value="")

In [11]:
df["trending_date"].sort_values().head(3)

3400    17.01.12
3528    17.01.12
3529    17.01.12
Name: trending_date, dtype: object

In [13]:
df["trending_date"].sort_values(ascending=False).head(3)

38144    18.31.05
38010    18.31.05
38020    18.31.05
Name: trending_date, dtype: object

Sorting "trending_date" with sort_values is not giving earliest and latest days as desired because format is year.day.month. Was able to use output from df["trending_date"].unique() to find earliest and latest days

In [21]:
df["trending_date"].unique().size

205

# '17.14.11' is earliest day. '18.14.06' is latest day.  
11/14/2017 --- 6/14/2018  Using time calculator: 213 days (7months)  
213-205=8 This means that 8 days are missing trending video data. (Given "trending_date" size above)

In [24]:
df["video_id"].unique().size

6351

# Videos appear multiple times. 6351 videos. 40949 rows

In [26]:
df.describe()

category_id         views         likes      dislikes  comment_count
count  40949.000000  4.094900e+04  4.094900e+04  4.094900e+04   4.094900e+04
mean      19.972429  2.360785e+06  7.426670e+04  3.711401e+03   8.446804e+03
std        7.568327  7.394114e+06  2.288853e+05  2.902971e+04   3.743049e+04
min        1.000000  5.490000e+02  0.000000e+00  0.000000e+00   0.000000e+00
25%       17.000000  2.423290e+05  5.424000e+03  2.020000e+02   6.140000e+02
50%       24.000000  6.818610e+05  1.809100e+04  6.310000e+02   1.856000e+03
75%       25.000000  1.823157e+06  5.541700e+04  1.938000e+03   5.755000e+03
max       43.000000  2.252119e+08  5.613827e+06  1.674420e+06   1.361580e+06

In [28]:
df["views"].sort_values(ascending=False).head()

38547    225211923
38345    220490543
38146    217750076
37935    210338856
37730    205643016
Name: views, dtype: int64

# View count range suprisingly large 549 - 225,211,923

In [31]:
df.describe(include = ['O'])

video_id trending_date                                       title  \
count         40949         40949                                       40949   
unique         6351           205                                        6455   
top     j4KvrAUjn6c      17.14.12  WE MADE OUR MOM CRY...HER DREAM CAME TRUE!   
freq             30           200                                          30   

       channel_title              publish_time    tags  \
count          40949                     40949   40949   
unique          2207                      6269    6055   
top             ESPN  2018-05-18T14:00:04.000Z  [none]   
freq             203                        50    1535   

                                        thumbnail_link description  
count                                            40949       40949  
unique                                            6352        6902  
top     https://i.ytimg.com/vi/j4KvrAUjn6c/default.jpg              
freq                                                30         570

# 6055 / 6351 unique tags to unique videos  
# The tag data has good variety and usefulness

In [36]:
df["tags"].unique()[25:30]

array(['spaghetti burrito|"diy burrito"|"spaghetti"|"burrito"|"burrito recipe"|"spaghetti deep fried"|"deep fried spaghetti"|"spgahetti sandwich"|"giant burrito"',
       'numberphile|"prime numbers"|"proth prime"',
       'Smart mug|"Heated thermos"|"tech"|"gift idea"|"unboxed"|"Ember mug"|"ember"|"Ember review"|"Heated mug"|"teardown"|"technology"|"Thermous"|"winter"|"holiday"|"christmas"|"gift"|"mug"|"unboxing"',
       'auth-jvardon-auth',
       'freaks and geeks|"jason segel"|"judd apatow"|"drums"|"rush"|"paul feig"|"drummer"|"nick andopolis"'],
      dtype=object)

In [43]:
df.corr()

category_id     views     likes  dislikes  \
category_id                1.000000 -0.168231 -0.173921 -0.033547   
views                     -0.168231  1.000000  0.849177  0.472213   
likes                     -0.173921  0.849177  1.000000  0.447186   
dislikes                  -0.033547  0.472213  0.447186  1.000000   
comment_count             -0.076307  0.617621  0.803057  0.700184   
comments_disabled          0.048949  0.002677 -0.028918 -0.004431   
ratings_disabled          -0.013506  0.015355 -0.020888 -0.008230   
video_error_or_removed    -0.030011 -0.002256 -0.002641 -0.001853   

                        comment_count  comments_disabled  ratings_disabled  \
category_id                 -0.076307           0.048949         -0.013506   
views                        0.617621           0.002677          0.015355   
likes                        0.803057          -0.028918         -0.020888   
dislikes                     0.700184          -0.004431         -0.008230   
comment_count                1.000000          -0.028277         -0.013819   
comments_disabled           -0.028277           1.000000          0.319230   
ratings_disabled            -0.013819           0.319230          1.000000   
video_error_or_removed      -0.003725          -0.002970         -0.001526   

                        video_error_or_removed  
category_id                          -0.030011  
views                                -0.002256  
likes                                -0.002641  
dislikes                             -0.001853  
comment_count                        -0.003725  
comments_disabled                    -0.002970  
ratings_disabled                     -0.001526  
video_error_or_removed                1.000000

Viewcount is understandably corellated to likes (0.849), dislikes (0.472), and comment count (0.618)

In [42]:
np.sort(df["category_id"].unique())

array([ 1,  2, 10, 15, 17, 19, 20, 22, 23, 24, 25, 26, 27, 28, 29, 43])